File: app/src/main/java/com/slapshotapps/dragonshockey/activities/schedule/adapters/ScheduleAdapter.java
package com.slapshotapps.dragonshockey.activities.schedule.adapters;

import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import com.slapshotapps.dragonshockey.R;
import com.slapshotapps.dragonshockey.Utils.DateFormaters;
import com.slapshotapps.dragonshockey.Utils.FormattingUtils;
import com.slapshotapps.dragonshockey.models.Game;
import com.slapshotapps.dragonshockey.models.SeasonSchedule;
import java.util.Calendar;
import java.util.Date;
import java.util.Locale;

/**
 *
 */
public class ScheduleAdapter extends RecyclerView.Adapter<ScheduleAdapter.GameViewHolder> {
    private SeasonSchedule schedule;

    public ScheduleAdapter(SeasonSchedule schedule) {
        this.schedule = schedule;
    }

    @Override
    public int getItemCount() {
        return schedule.numberOfGames();
    }

    @Override
    public GameViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {

        View gameView = LayoutInflater.from(parent.getContext())
            .inflate(R.layout.view_game_details, parent, false);

        return new GameViewHolder(gameView);
    }

    @Override
    public void onBindViewHolder(GameViewHolder holder, int position) {
        Game game = schedule.getAllGames().get(position);

        holder.setGameDate(game.gameTimeToDate());
        holder.setGameOpponent(game.getOpponent());
        holder.setGameResult(FormattingUtils.getGameScore(game.getGameResult(), game.getOpponent()));
    }

    public static class GameViewHolder extends RecyclerView.ViewHolder {
        private TextView gameDay;
        private TextView gameTime;
        private TextView opponent;
        private TextView gameResult;

        public GameViewHolder(View itemView) {
            super(itemView);

            gameDay = (TextView) itemView.findViewById(R.id.game_day);
            gameTime = (TextView) itemView.findViewById(R.id.game_time);
            opponent = (TextView) itemView.findViewById(R.id.opponent);
            gameResult = (TextView) itemView.findViewById(R.id.game_result);
        }

        public void setGameDate(Date gameDate) {
            Calendar calendar = Calendar.getInstance();
            calendar.setTime(gameDate);

            final String gameDayStr =
                calendar.getDisplayName(Calendar.DAY_OF_WEEK, Calendar.SHORT, Locale.US)
                    + " "
                    + calendar.getDisplayName(Calendar.MONTH, Calendar.SHORT, Locale.US)
                    + " "
                    + FormattingUtils.getValueWithSuffix(calendar.get(Calendar.DAY_OF_MONTH));

            gameDay.setText(gameDayStr);
            gameTime.setText(DateFormaters.getGameTime(gameDate));
        }

        public void setGameOpponent(String opponent) {
            this.opponent.setText(opponent);
        }

        public void setGameResult(String result) {
            gameResult.setText(result);
        }
    }
}

File: app/src/main/java/com/slapshotapps/dragonshockey/activities/schedule/adapters/ScheduleAdapter.kt
package com.slapshotapps.dragonshockey.activities.schedule.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.slapshotapps.dragonshockey.R
import com.slapshotapps.dragonshockey.databinding.ViewGameDetailsBinding
import com.slapshotapps.dragonshockey.utils.DateFormatter
import com.slapshotapps.dragonshockey.utils.FormattingUtils
import com.slapshotapps.dragonshockey.models.SeasonSchedule
import java.util.*

/**
 *
 */
class ScheduleAdapter(
        private val schedule: SeasonSchedule) : RecyclerView.Adapter<ScheduleAdapter.GameViewHolder>() {

    override fun getItemCount(): Int {
        return schedule.numberOfGames()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): GameViewHolder {

        val binding = ViewGameDetailsBinding.inflate(LayoutInflater.from(parent.context), parent, false)

        return GameViewHolder(binding)
    }

    override fun onBindViewHolder(holder: GameViewHolder, position: Int) {
        val game = schedule.allGames[position]

        holder.setGameDate(game.gameTimeToDate())
        holder.setGameOpponent(game.opponent)
        holder.setGameResult(FormattingUtils.getGameScore(game.gameResult, game.opponent))
    }

    class GameViewHolder(private val binding: ViewGameDetailsBinding) : RecyclerView.ViewHolder(binding.root) {

        fun setGameDate(gameDate: Date?) {
            val calendar = Calendar.getInstance()
            if (gameDate != null) {
                calendar.time = gameDate

                val gameDayStr = (calendar.getDisplayName(Calendar.DAY_OF_WEEK, Calendar.SHORT, Locale.US)
                        + " "
                        + calendar.getDisplayName(Calendar.MONTH, Calendar.SHORT, Locale.US)
                        + " "
                        + FormattingUtils.getValueWithSuffix(calendar.get(Calendar.DAY_OF_MONTH)))

                binding.gameDay.text = gameDayStr
                binding.gameTime.text = DateFormatter.getGameTime(gameDate)
            }
        }

        fun setGameOpponent(opponent: String?) {
            binding.opponent.text = opponent
        }

        fun setGameResult(result: String) {
            binding.gameResult.text = result
        }
    }
}
